modules/samples/extensions/geo-event-fuser/src/main/java/org/wso2/siddhi/extension/geo/geoeventfuser/FuseEvents.java

```diff
@@ -100,46 +99,38 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
         nextProcessor.process(streamEventChunk);
     }
 
+    /**
+     * Get number of deployed execution plans
+     *
+     * @return number of deployed execution plans
+     */
     public Integer getDeployedExecutionPlansCount() {
         return ExecutionPlansCount.getNumberOfExecutionPlans();
     }
 
+    /**
+     * Fuse given {@link StreamEvent}.
+     * -Precedence of states when fusing-
+     * goes low to high from LHS to RHS
+     * OFFLINE < NORMAL < WARNING < ALERT
+     * States will be in all caps to mimics
+     * that states not get on the way
+     *
+     * @param event {@link StreamEvent} to fuse
+     */
     public void fuseEvent(StreamEvent event) {
-
-    /*
-        * --For reference--
-        * -Precedence of states-
-        * goes low to high from LHS to RHS
-        *
-        * OFFLINE < NORMAL < WARNING < ALERT
-        * */
-
-
-    /*
-        * --For reference--
-        *   Higher the index higher the Precedence
-        *   States are in all caps to mimics that states not get on the way
-        *
-        * */
-
-        String[] statesArray = new String[] { "OFFLINE", "NORMAL", "WARNING", "ALERTED" };
+        String eventId = (String) event.getOutputData()[eventIdPosition];
+        ArrayList<StreamEvent> receivedEvents = eventsBuffer.get(eventId);
```

Comment:
Use the interface List as the type of the variable instead of the implementation class as a general practice (unless any specific methods of the implementation class is needed)